pkgs/tools/misc/betterdiscordctl/default.nix

```
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, pkgs }:
+
+stdenv.mkDerivation rec {
+  pname = "betterdiscordctl";
+  version = "1.6.1";
+
+  src = fetchFromGitHub {
+    owner = "bb010g";
+    repo = "betterdiscordctl";
+    rev = "v${version}";
+    sha256 = "0qbas7gbjpi688vmsq4yjbycc0g84zkilsfiikvjmj3nfcmy2xh6";
+  };
+
+  patchPhase = ''
+    substituteInPlace betterdiscordctl --replace "/opt" "${pkgs.discord}/opt"
```

The self-upgrade mechanism should be disabled.
```suggestion
    substituteInPlace betterdiscordctl --replace "/opt" "${pkgs.discord}/opt"
    substituteInPlace betterdiscordctl --replace "DISABLE_UPGRADE=" "DISABLE_UPGRADE=1"
```

---

> The self-upgrade mechanism should be disabled.

Will do :+1: 

> Also, will ${pkgs.discord} work if Discord isn't installed before betterdiscordctl?

It fetches Discord.

> And what about PTB and Canary?

Look like we need to do something like [Discord itself](https://github.com/NixOS/nixpkgs/tree/bd4145016d137310be083a5e11062fdd2e07ba77/pkgs/applications/networking/instant-messengers/discord) to support that. I'll work on that in a bit.